File: app/src/main/java/de/heilsen/ganzhornfest/GanzhornfestApplication.java
package de.heilsen.ganzhornfest;

import android.app.Application;

import de.heilsen.ganzhornfest.di.ServiceLocator;
import de.heilsen.ganzhornfest.interactor.ThreadedGetClubList;
import de.heilsen.ganzhornfest.interactor.ThreadedShowClubDetail;


public class GanzhornfestApplication extends Application {
    @Override
    public void onCreate() {
        super.onCreate();
        ServiceLocator.load(new ServiceLocator(
                //new GetClubList(new FakeClubRepository())));
                new ThreadedGetClubList(new FakeClubRepository()), new ThreadedShowClubDetail(null)));

    }
}

File: app/src/main/kotlin/de/heilsen/ganzhornfest/GanzhornfestApplication.kt
package de.heilsen.ganzhornfest

import android.app.Application
import com.squareup.anvil.annotations.ContributesTo
import de.heilsen.ganzhornfest.di.AppComponent
import de.heilsen.ganzhornfest.di.AppComponentProvider
import de.heilsen.ganzhornfest.di.AppScope
import de.heilsen.ganzhornfest.di.DaggerAppComponent
import de.heilsen.ganzhornfest.di.appScope
import de.heilsen.ganzhornfest.di.getValue
import timber.log.Timber
import java.security.KeyStore.Entry
import javax.inject.Inject

@ContributesTo(AppScope::class)
interface EntryPoint {
    fun inject(ganzhornfestApplication: GanzhornfestApplication)
}

class GanzhornfestApplication : Application(), AppComponentProvider {

    override val appComponent: AppComponent by lazy { DaggerAppComponent.factory().create(this) }

    @Inject
    lateinit var timberTrees: Set<@JvmSuppressWildcards Timber.Tree>

    override fun onCreate() {
        super.onCreate()
        val entrypoint: EntryPoint by appScope
        entrypoint.inject(this)
        Timber.plant(*timberTrees.toTypedArray())
    }
}
